### ISSUE TITLE
Results reproduction

### ISSUE THREAD

**Issue by hayk-ghazaryan** [matterport/Mask_RCNN#49]
Hi, thanks a lot for publishing this code as open source project.

I trained the net without doing any changes to the code/configuration, and by initializing it with ImageNet weights. After training the weights output is different and gives worse results(but somewhat similar) compared to the one which was provided in the repository.
 I have some thoughts on why the train could go wrong, although I’m not sure in my correctness. From the code, it seems to me that the mean is applied at the end of loss functions calculations and results from different GPUs are concatenated during the model parallelization, hence no need learning rate change. I have trained the net with 1 GPU, and as far I understand in such case in each step the net is trained with 2 images, but if the net in trained with 8 GPUs, then in each step the net will be trained with 16 images, and my thought is that in that case the train will be more stable, since the impact of “noise” will be smaller because the gradient direction will be determined by using 16 images instead of 2. 
Please correct me if you think I made a wrong conclusions, and do you have any ideas on why the train could go wrong? 

Thanks in advance!

Here is an inference result with my trained weights.

![result_1](https://user-images.githubusercontent.com/27297512/32694512-1211e7d4-c75b-11e7-99f3-ae69fb2d27c1.png)

![tensor_board](https://user-images.githubusercontent.com/27297512/32694527-9291fdcc-c75b-11e7-8e87-a08b898f4826.png)

**Comment by IQ17** [comment]
Hi there is a big jump in your loss curve, which appears in my case also. Have you got any idea what happened there, because all other curves are decreasing. Thanks!

**Comment by IQ17** [comment]
By the way, I am a novel, could you tell me where to get a copy the pretrained imagenet weight? Thanks!

**Comment by waleedka** [comment]
The batch size does affect the training. I didn't do a through test to measure the effect, but I noticed once that the loss function settled while training on 1 GPU and then when I continued the training on 8 GPUs the loss got to lower values. It could've been a coincidence or I changed something else, I didn't repeat the test to verify. But, intuitively, the batch size should improve the training. I considered updating the training code such that when training on 1 GPU it averages the gradients of 8 batches and then apply them together. This should emulate training on 8 GPUs. But I didn't get around to doing that. 

As for the bump in the loss value. I just pushed a fix for that. The bump happens because I added the regularization loss from trainable layers, so as you transition from stage 1 to stage 2 training and train more layers the loss value increases. The fix shouldn't affect the training results, but it makes the loss graph look nicer.

**Comment by IQ17** [comment]
Hi I also trained from the pretrained imagenet model with 1 GPU, below are my figures. 
By the way, should I stop training at 40 iterations, because the val_mrcnn_class_loss jumps up...but others are decreasing, any suggestion on what might happened? Thanks!

![screenshot-2017-11-13 tensorboard](https://user-images.githubusercontent.com/5336719/32706151-1404c150-c856-11e7-8fc7-55944d6e2f76.png)

**Comment by hayk-ghazaryan** [comment]
In my opinion, the above  waleedka's comment contains the answer.

It is just regularization losses.
First stage trains only net heads, hence body regularization losses absent there.

As I saw training losses are basically identical to mine.

**Comment by Cyndishine** [comment]
Hi! How did you observe the loss curve by tensorboard? I just can observe the structure of the net in 'graph', but can't see any loss curve in 'scalar'. Is There any modification should I do with the code?

**Comment by gsujansai** [comment]
@Cyndishine I can see the loss curve in scalars just by using tensorboard --logdir=logs/path to logs. No modifications to code required.

**Comment by ccpocker** [comment]
@Cyndishine I meet the same problem as yours. Do you solve it?
